chia/simulator/simulator_test_tools.py

```diff
@@ -154,3 +165,5 @@ async def get_full_chia_simulator(
 
         await ws_server.stop()
         await shutdown_event.wait()  # wait till shutdown is complete
+        if temp_dir is not None:
+            temp_dir.cleanup()
```

Comment:
Context manager please.  It's a touch more complicated since you only conditionally need this, but you have two basic choices.  Either conditionally choose the manager from either https://docs.python.org/3.10/library/contextlib.html#contextlib.nullcontext or the temporary directory if you need it and then always enter the chosen manager, or you can always enter https://docs.python.org/3.10/library/contextlib.html#contextlib.ExitStack and then only conditionally have it enter the temporary directory manager.

Or...  you skip this complexity and just make the caller give you a directory to work in.  It's not that hard for them to make a temp dir if that's where they want this function to operate.  It looks like we only call this one place and we pass a root there so the optionality is entirely, well, optional.

Reply:
It seems to me that two lines of manual cleanup is preferable to more complicated code simply to use a context manager. but I'll consider further changes....

Reply:
There's `try:`/`finally:` as well, but the ugliness suggests to me that making the caller do it if needed is the clean solution.

Reply:
Rearranged the code to make the caller specify it, which in our test code was simplified by using tmp_path fixture